integration_tests/src/test_util.rs

```diff
@@ -125,22 +125,22 @@ impl log::Log for FileLogger {
 /// The suite of tests to run against a fully-instantated and storage-backed directory.
 /// This will publish 3 epochs of ```num_users``` records and
 /// perform 10 random lookup proofs + 2 random history proofs + and audit proof from epochs 1u64 -> 2u64
+/// Note that for `num_users` < 2, there are still exactly two users created and ran
 pub(crate) async fn directory_test_suite<S: akd::storage::Storage + Sync + Send>(
     mysql_db: &S,
     num_users: usize,
 ) {
     // generate the test data
     let mut rng = thread_rng();
 
-    let mut users: Vec<String> = vec![];
-    for _ in 0..num_users {
-        users.push(
-            thread_rng()
-                .sample_iter(&Alphanumeric)
-                .take(30)
-                .map(char::from)
-                .collect(),
-        );
+    let mut users: Vec<Vec<u8>> = vec![];
+
+    // ensure that one user is string bytes and the other is bytes that are not utf-8
+    users.push(Vec::from("a username that is valid utf-8"));
+    users.push(vec![0x80]); // a username that is NOT valid utf-8
+
+    for _ in 2..num_users {
+        users.push((&mut rng).sample_iter(&Standard).take(30).collect());
```

Comment:
For this particular test, I was having an issue wherein I'd updated the wrong database such that the test was still trying to put this value to a `VARCHAR` when I thought I'd already updated it to a `VARBINARY`. The MySQL parameter replacement would quietly try to coerce the value to a string.

However, this test and a bunch of other tests in this project are generating random data. This meant that sometimes this test would _work_ because the random bytes it happened to have generated were legal UTF-8. Since this was the test that was going sideways, you can see I've forced a case with "bad" UTF-8 to be certain.

Note that there is no such guard on other tests that are generating random data. It's unlikely they all happen to be hitting the "sweet spot" all the times I've run them, but it's worth keeping in mind.

Similarly, there's a vanishingly small chance that two "users" we be generated with the same key, both here and in other tests. Would the test fail? Would we be able to diagnose what happened? I didn't think this was such a big deal that it was worth addressing in this PR.